test/Rx/Functional/Operator/ZipTest.php

```diff
@@ -0,0 +1,615 @@
+    }
+
+    public function testZipNeverNonEmpty()
+    {
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(220)
+        ]);
+        $e2 = new NeverObservable();
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([], $results->getMessages());
+    }
+
+    public function testZipNonEmptyNever()
+    {
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(220)
+        ]);
+        $e2 = new NeverObservable();
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([], $results->getMessages());
+    }
+
+    public function testZipNonEmptyNonEmpty()
+    {
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(230)
+        ]);
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(220, 3),
+            onCompleted(240)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onNext(220, 2 + 3),
+            onCompleted(240)
+        ], $results->getMessages());
+    }
+
+    public function testZipEmptyError()
+    {
+        $error = new \Exception();
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(230)
+        ]);
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onError(220, $error)
+        ], $results->getMessages());
+    }
+
+    public function testZipErrorEmpty()
+    {
+        $error = new \Exception();
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(230)
+        ]);
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onError(220, $error)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($e1, $e2) {
+            return $e1->zip([$e2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([
+            onError(220, $error)
+        ], $results->getMessages());
```

Comment:
This is the same as the test above? Even if you would flip the sequences, does it add many value over the other test? Perhaps we should just remove it?